Update NanoUI

neersighted opened the pull request:
This PR ports the new NanoUI code from Baystation12. While mostly under the hood, this will enable me to update and refactor our existing NanoUIs, and later NanoUI 'everything' on /tg/.

So far I have ported the Baycode, updated the libraries, refactored the JSON code, all the interfaces, implemented asset_cache, and re-factored the NanoUI API.

Eventually I will write a guide and NanoUI all the things -- however, these will be in separate PRs.

Some recent screenshots (unsorted):
![2015-11-27_15-24-18](https://butt.githubusercontent.com/assets/1049222/11450365/94babf68-956b-11e5-8df4-5455d57240a8.png)
![2015-11-27_13-30-48](https://butt.githubusercontent.com/assets/1049222/11450369/94bd45c6-956b-11e5-995f-1ef01345722d.png)
![2015-11-27_22-59-58](https://butt.githubusercontent.com/assets/1049222/11450371/94c47dc8-956b-11e5-9e66-7c6c9254863f.png)
![2015-11-27_22-45-47](https://butt.githubusercontent.com/assets/1049222/11450373/94c8a9a2-956b-11e5-9ed7-20a548b68d71.png)
![2015-11-27_17-16-27](https://butt.githubusercontent.com/assets/1049222/11450375/94caae00-956b-11e5-9b3d-95049c2b05fe.png)
![2015-11-27_16-36-55](https://butt.githubusercontent.com/assets/1049222/11450372/94c7f50c-956b-11e5-918e-6f916697ce72.png)
![2015-11-27_11-44-35](https://butt.githubusercontent.com/assets/1049222/11450382/c637e85e-956b-11e5-8dbd-8d47cd7fc5cc.png)
![2015-11-27_00-51-31](https://butt.githubusercontent.com/assets/1049222/11450383/d0196ba4-956b-11e5-9ee6-1db2b0477356.png)
![2015-11-26_11-59-15](https://butt.githubusercontent.com/assets/1049222/11450385/dc0338e6-956b-11e5-820d-91fa332ae23a.png)
![2015-11-26_11-53-09](https://butt.githubusercontent.com/assets/1049222/11450386/dc052d22-956b-11e5-87e5-038dcc5baf2a.png)

neersighted committed:
Update NanoUI

* Update NanoUI to the latest version, ported from Baystation12
* Update the Air Alarm UI
* Make changes to subsystems and some machines to make the new NanoUI
work

YotaXP commented:
You beautiful person, you.  Can we get a screenshot of any new UIs?

neersighted commented:
This is mostly under the hood, and I'm off to dinner. However, I have this...
![Air Alarm](https://i.imgur.com/BeznfLE.png)

Anonus commented:
I notice some images for syndicate interfaces, would that mean upgrading the syndicate telecrystal consoles and shuttle movement computer might also happen?

optimumtact commented:
:+1: on the condition that you make the json reader/writer procs fully typed again, as per our coding stnadards.

Shadowlight213 commented:
:+1:×10000000 so much yes.

duncathan commented on code/modules/nano/JSON Reader.dm:
```diff
@@ -20,185 +20,186 @@
-					val += char
+	proc
+		// scanner
+		ScanJson(json)
```
Yeah definitely fix the relative pathing here, like oranges pointed out

Shadowlight213 commented on code/modules/mob/living/silicon/ai/ai.dm:
```diff
@@ -795,6 +795,7 @@ var/list/ai_list = list()
 
 	if(stat == 2)
 		return //won't work if dead
+
```
Why is there a random break here reeee

neersighted replied:
Because whitespace damnit.

optimumtact replied:
agreed

optimumtact commented on code/ATMOSPHERICS/components/unary_devices/cryo.dm:
```diff
@@ -203,22 +203,18 @@
 	if(href_list["switchOff"])
 		on = 0
 
+	if(href_list["ejectOccupant"])
```
One other thing, did you test cyro still works after these topic changes?

neersighted replied:
Yep, checked it and it works great. Only tested with a mindless `/mob/living/carbon/human` though.

neersighted committed:
Refactor JSON

* Move into seperate module
* Use aboslute pathing

neersighted committed:
Remove Bay's minimap-generating code.

We have our own in the crew computer we will use.

neersighted committed:
Add NanoUI debug verb

A quick tool that helps me reload templates.
Also refactor NanoUI resource code a bit.

neersighted committed:
Minor template tweaks

neersighted committed:
Refactor NanoUI API

Also update NanoUI-using code to the new API.

neersighted committed:
Fix NanoUI debug verb

neersighted committed:
Correct several oversights

Aranclanos commented:
>nanoui pr
>is not removing it

:-1:

Iamgoofball commented:
fuck off

:+1:

neersighted committed:
Further re-work APCs and Air Alarms

neersighted committed:
Further UI Improvements

Finish redesign of APC and Air Alarm

Supermichael777 commented:
hope this fixes my weird clientside issues with nano in 508+

neersighted committed:
Finish refactor/retooling of NanoUI core

Also:
* Fix NanoUI on IE11+
* Add missing files (oops!)

Supermichael777 commented:
uncheck file dr

neersighted committed:
Update NanoUI Javascript libraries, SMES UI

neersighted committed:
Fix NanoUIs flashing, update Tanks, Solars, and Canisters

neersighted committed:
Merge remote-tracking branch 'tgstation/master' into nanoui_update

neersighted committed:
Update Cryo, NanoUI states, add new contains() helper

neersighted committed:
Update the rest of the Atmos stuff

neersighted committed:
Update NanoUI to use the asset_cache system

Also move a bunch of folders around.

neersighted commented:
@Supermichael777 It's pretty close. Maybe by tomorrow?

MrStonedOne commented:
Seems fine to me.

MrStonedOne commented:
So far no bugs reported

This can be merged

Shadowlight213 commented:
It's happening!
